File: CortexSuite/vision/benchmarks/localization/src/matlab/generateSample.m
function [retQuat retVel retPos]=generateSample(w, quat, vel, pos, STDDEV_VEL, STDDEV_POS)
sampledXId=weightedSample(w);

rows = size(sampledXId, 1);
cols = size(sampledXId, 2);

if(cols > 1)
    disp(123456);
end

% retQuat = zeros(rows, 1);
% retVel = zeros(rows, 1);
% retPos = zeros(rows, 1);

% for i=1:rows
%     retQuat(i,1) = quat(sampleXId(i,1),1);
%     retVel(i,1) = vel(sampleXId(i,1),1) + randnWrapper(1,1) * STDDEV_VEL;
%     retPos(i,1) = pos(sampleXId(i,1),1) + randnWrapper(1,1) * STDDEV_POS;    
% end

retQuat=quat(sampledXId,:);
retVel=vel(sampledXId,:);%+randnWrappern(n,3)*STDDEV_VEL;
retPos=pos(sampledXId,:);%+randnWrappern(n,3)*STDDEV_POS;

File: CortexSuite/vision/benchmarks/localization/src/c/generateSample.c
/********************************
Author: Sravanthi Kota Venkata
********************************/

#include <stdio.h>
#include <stdlib.h>
#include "localization.h"

void generateSample(F2D *w, F2D *quat, F2D *vel, F2D *pos)
{
    int rows, cols, i, j, index;
    I2D *sampleXId;
    F2D *retQuat, *retVel, *retPos;

    sampleXId = weightedSample(w);

    rows = sampleXId->height;
    cols = sampleXId->width;

    if(cols > 1)
        printf("ERROR: Cols more than 1.. Handle this case \n");

    retQuat = fSetArray(quat->height, quat->width, 0);
    retVel = fSetArray(vel->height, vel->width, 0);
    retPos = fSetArray(pos->height, pos->width, 0);

    for(i=0; i<rows; i++)
    {
        index = asubsref(sampleXId, i) - 1;
        for(j=0; j<quat->width; j++)
        {
            subsref(retQuat,i,j) = subsref(quat,index,j);
        }
    }

    for(i=0; i<rows; i++)
    {
        index = asubsref(sampleXId, i) - 1;
        for(j=0; j<vel->width; j++)
        {
            subsref(retVel,i,j) = subsref(vel,index,j);
        }
    }
    
    for(i=0; i<rows; i++)
    {
        index = asubsref(sampleXId, i) - 1;
        for(j=0; j<pos->width; j++)
        {
            subsref(retPos,i,j) = subsref(pos,index,j);
        }
    }

    for(i=0; i<quat->height; i++)
    {
        for(j=0; j<quat->width; j++)
        {
            subsref(quat,i,j) = subsref(retQuat,i,j);
        }
    }

    for(i=0; i<vel->height; i++)
    {
        for(j=0; j<vel->width; j++)
        {
            subsref(vel,i,j) = subsref(retVel,i,j);
        }
    }
    
    for(i=0; i<pos->height; i++)
    {
        for(j=0; j<pos->width; j++)
        {
            subsref(pos,i,j) = subsref(retPos,i,j);
        }
    }
    
    fFreeHandle(retQuat);
    fFreeHandle(retVel);
    fFreeHandle(retPos);
    iFreeHandle(sampleXId);
    
    return;
}




